brk.c

```diff
@@ -47,20 +47,15 @@
 #define HEAP_LEN   PAGES2BYTES(CONFIG_APPELFLOADER_BRK_NBPAGES)
 
 /*
- * For now we only support one custom stack
+ * For now we only support one custom heap
  */
 static void *base = NULL;
+static void *brk_cur = NULL;
 static void *zeroed = NULL;
 static intptr_t len = 0;
 
 UK_LLSYSCALL_R_DEFINE(void *, brk, void *, addr)
 {
-	if (!addr && base) {
-		/* another brk context request although we have one already */
-		uk_pr_crit("Cannot handle multiple user space heaps: Not implemented!\n");
-		return ERR2PTR(-ENOMEM);
-	}
-
 	if (!addr) {
```

Comment:
🤔  Because the previous condition is removed, we should check here for `base` because we need to catch the state that no heap is allocated yet and we don't want to allocate a new heap region always when addr is set to `0`. I am thinking of something like this:
```c
if (!base) {
        base = uk_palloc(uk_alloc_get_default(), HEAP_PAGES);
        if (!base)
                return ERR2PTR(-ENOMEM);

        /* initialize brk_cur with start of allocated heap region */
        brk_cur = base;
}
```

Reply:
I added the check. Thank you.